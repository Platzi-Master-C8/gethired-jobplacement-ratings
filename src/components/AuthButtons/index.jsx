import React from 'react';

import Button from '@mui/material/Button';

import { goToEnterprise } from '../../utils';

export const AuthenticationButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Log in
        </Button>
    );
};

export const SignupButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Sign up
        </Button>
    );
};
