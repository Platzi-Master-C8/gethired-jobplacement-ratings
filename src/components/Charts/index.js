import RadarChart from "./RadarChart";
import LineChart from "./LineChart";
export { RadarChart, LineChart }