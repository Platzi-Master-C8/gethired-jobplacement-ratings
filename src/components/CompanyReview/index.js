import CompanyReview from './CompanyReview';

export { CompanyReview };
