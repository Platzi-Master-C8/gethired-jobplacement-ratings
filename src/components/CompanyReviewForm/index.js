import CompanyReviewForm from './CompanyReviewForm';

export { CompanyReviewForm };
