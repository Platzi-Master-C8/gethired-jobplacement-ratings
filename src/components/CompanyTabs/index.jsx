import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Tabs, Tab, Box } from '@mui/material';

const TabPanel = ({ children, value, index }) => {
    return (
        <div
            role="tabpanel"
            hidden={value !== index}
            id={`simple-tabpanel-${index}`}
            aria-labelledby={`simple-tab-${index}`}
        >
            {value === index && <Box sx={{ p: 3 }}>{children}</Box>}
        </div>
    );
};

TabPanel.propTypes = {
    children: PropTypes.node.isRequired,
    index: PropTypes.number.isRequired,
    value: PropTypes.number.isRequired,
};

const CompanyTabs = ({ tabsOptions }) => {
    const [value, setValue] = useState(1);

    const handleChange = (_event, newValue) => {
        setValue(newValue);
    };

    return (
        <Box sx={{ width: '100%' }}>
            <Box>
                <Tabs value={value} onChange={handleChange}>
                    {tabsOptions.map((option) => (
                        <Tab key={`TabButton-${option.tabKey}`} label={option.tabKey} />
                    ))}
                </Tabs>
            </Box>
            {tabsOptions.map((option, i) => (
                <TabPanel key={`TabContent-${option.tabKey}`} value={value} index={i}>
                    {option.tabContent}
                </TabPanel>
            ))}
        </Box>
    );
};

CompanyTabs.propTypes = {
    tabsOptions: PropTypes.arrayOf(
        PropTypes.shape({
            tabKey: PropTypes.string.isRequired,
            tabContent: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
        }),
    ).isRequired,
};

export default CompanyTabs;
