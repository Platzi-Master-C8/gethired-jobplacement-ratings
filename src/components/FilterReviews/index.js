import FilterReviews from './FilterReviews';

export {
    FilterReviews
}