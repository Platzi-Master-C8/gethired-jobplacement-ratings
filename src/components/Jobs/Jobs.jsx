import React from 'react';
import { ReviewTheApplicantForm } from '../ReviewTheApplicantForm';
import { ReviewApplicationProcessForm } from '../ReviewApplicationProcess';

const Jobs = () => (
    <div>
        <ReviewTheApplicantForm company_id={1} applicant_id={1} />
        <ReviewApplicationProcessForm company_id={1} applicant_id={1} />
    </div>
);

export default Jobs;
