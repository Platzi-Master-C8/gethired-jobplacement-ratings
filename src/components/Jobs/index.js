import Jobs from './Jobs';

export { Jobs };