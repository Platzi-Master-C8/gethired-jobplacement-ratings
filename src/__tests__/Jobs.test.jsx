import React from 'react';
import { render } from '@testing-library/react';
import { Jobs } from '../components/Jobs';

describe('component <Jobs/>', () => {
    test('renders the Jobs Component', () => {
        const { asFragment } = render(<Jobs />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the buttons', () => {
        const component = render(<Jobs />);
        const buttons = component.container.querySelectorAll('button');

        expect(buttons.length).toBe(2);
    });
});
