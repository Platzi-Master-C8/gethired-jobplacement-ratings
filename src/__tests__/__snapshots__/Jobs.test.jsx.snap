// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Jobs/> renders the Jobs Component 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12 css-hagxv6-MuiGrid-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Review the applicant 
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div />
    </div>
    <div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12 css-hagxv6-MuiGrid-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Review your application process
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div />
    </div>
  </div>
</DocumentFragment>
`;
